dashboard/test/integration/stage_lock_test.rb

```diff
@@ -0,0 +1,80 @@
+require 'test_helper'
+
+class StageLockTest < ActionDispatch::IntegrationTest
+  include LevelsHelper # for build_script_level_path
+
+  setup_all do
+    @student = create :student
+    @teacher = create :authorized_teacher
+    @section = create :section, user_id: @teacher.id
+    Follower.create!(section_id: @section.id, student_user_id: @student.id, user: @teacher)
+
+    @script = create :script
+    @lockable_stage = create(:stage, script: @script, name: 'Lockable Stage', lockable: true)
+    external = create(:external, name: 'markdown level')
+    @lockable_external_sl = create(:script_level, script: @script, stage: @lockable_stage, levels: [external])
+    @level_group = create(:level_group, name: 'assessment 1')
+    @lockable_level_group_sl = create(:script_level, script: @script, stage: @lockable_stage, levels: [@level_group], assessment: true)
+  end
+
+  test 'authorized teacher viewing lockable stage contents' do
+    sign_in @teacher
+
+    get build_script_level_path(@lockable_external_sl)
+    assert_response :success
+    assert_includes response.body, 'lorem ipsum'
+    assert_select "#locked-stage", 1
+    # data-hidden indicates that the client will decide whether the teacher
+    # will see the locked-stage message via the ViewAsToggle.
+    assert_select "#locked-stage[data-hidden]", 1
+
+    # This needs to be an integration test rather than a controller test in
+    # order to follow the redirect which adds the /page/1 suffix.
+    get build_script_level_path(@lockable_level_group_sl)
+    assert_response :redirect
+    get URI.parse(@response.redirect_url).path
```

Comment:
there is a `follow_redirect!` method available in integration tests that you should be able to use here. here's an example:

https://github.com/code-dot-org/code-dot-org/blob/94e720152b82437ac9bfdf85c9a1d1a082130b04/dashboard/test/controllers/omniauth_callbacks_controller/google_oauth2_test.rb#L53-L57

Reply:
Good call Maddie! That's much better. 

Incidentally, in looking for this method, I discovered that we have a many instances of `ActionDispatch::IntegrationTest` in the `dashboard/test/controllers` directory. I'm not sure how conventional our `dashboard/test` directory layout is, but what we currently have seems to suggests that any new instances of `ActionDispatch::IntegrationTest` should go in `dashboard/test/integration`, not `dashboard/test/controllers`. Just a thing to keep an eye out for in code reviews I guess.